File: FinanceTracker/Models/UserSettingsModel.cs
using FinanceTracker.Core.Models;

namespace FinanceTracker.Models
{
    public class UserSettingsModel
    {
        public UserSettingCode SettingCode { get; set; }
        public ToolCode? ToolCode { get; set; }
        public TileCode? TileCode { get; set; }
    }
}

File: ClientApp/src/app/core/models/user-settings/user-settings-model.ts
import { TileCode } from "../tile-code";
import { ToolCode } from "../tool-code";
import { UserSettingCode } from "../user-setting-code";

export interface UserSettingsModel {
    settingCode: UserSettingCode;
    toolCode?: ToolCode;
    tileCode?: TileCode;
}